### ISSUE TITLE
Auto group column layout issue caused by async setAutoGroupColumnDef

### ISSUE THREAD

**Issue by YuerLee** [ag-grid/ag-grid#5689]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => see 'Providing a Reproducible Scenario'
[ ] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[ ] support request => see 'Requesting Community Support'
```

**Current behavior**
Async setAutoGroupColumnDef, the auto group column will have an incorrect `left` style since v28.2.0. 

Reproduced in v28.2.0 vs v28.1.1: https://stackblitz.com/edit/web-platform-6xxacr?file=main.js

**Expected behavior**
Keep the normal layout.

**Please tell us about your environment:**
stackblitz on macOS.

* **AG Grid version:**
28.2.0

* **Browser:**
'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/106.0.0.0 Safari/537.36'

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
ES6/7

**Comment by StephenCooper** [comment]
Hello,

Thank you for reporting this - much appreciated!

Based on your description we've now reproduced this issue as a bug.

We have added this bug to our backlog and we are tracking it with the following reference and description:
AG-7420 Auto group column layout issue caused by async setAutoGroupColumnDef

We try to fix bugs from one release to the next, so this should be fixed in the next release or the one after if it was raised too close to the next release date.

See whether this item will be part of our next release by checking the NEXT RELEASE checkbox on the product pipeline page:
https://www.ag-grid.com/ag-grid-pipeline/

Now that this issue is recorded in our backlog, we will soon close the GitHub issue because we will not update it when this bug is fixed in a future version of AG Grid. The best way to get an update on this item is to sign up for AG Grid new release notifications. Once a new AG Grid version is released, look for the item reference above (AG-7420 ) on the changelog page to see if it's resolved in that version:
https://www.ag-grid.com/ag-grid-changelog/

Thanks again for bringing this up with us - we appreciate your time!
